### ISSUE TITLE
More warnings

### ISSUE THREAD

**Issue by hadley** [r-lib/rcmdcheck#32]
It would be great (if possible) to make the default warnings more stringent:

```R
options(
  warn = 1,
  warnPartialMatchDollar = TRUE,
  warnPartialMatchArgs = TRUE,
  warnPartialMatchAttr = TRUE
)
```

See https://github.com/hadley/devtools/issues/965 for more discussion

**Comment by gaborcsardi** [comment]
It is not entirely clear how this could be implemented. Ideally we only want to use it for package code, i.e. examples and tests, but `R CMD check` does not give us hooks to do so, and it also starts up separate processes for examples and tests, which makes setting the option somewhat more cumbersome.

Maybe we could create a fake profile that sets the options, but then they will apply to the base R check code as well. To restrict it to tests, we could look at the `R_TESTS` env var, but then we might as well set these options in testthat. I don't know if there is a way to detect if `R CMD check`  is running  examples in the current session, maybe.

**Comment by gaborcsardi** [comment]
@hadley ?  @jimhester ?

**Comment by jimhester** [comment]
I think the best we could do is run the check with a temporary R profile, like we did in remotes.

As you said this would set it for the checking code as well, are there false positives in the base R code we would need to filter out?

**Comment by gaborcsardi** [comment]
There might be, yes, but how many, I cannot tell. I can add this now, and maybe make it opt-in?

rcmdcheck already uses a fake profile, but it does it wrong (does not include the contents of the original  profile), so I'll fix that as well.

**Comment by hadley** [comment]
Opt-in seems reasonable to me.

**Comment by gaborcsardi** [comment]
Seems like the process that runs the examples in `R CMD check` does not read the profiles, so we can't  inject these options in a fake profile, at least not into the examples run.

And then for the tests, it is probably easier if testthat just sets them?

**Comment by hadley** [comment]
Yeah, makes sense. Feel free to close this issue.
